SU2_CFD/include/variables/CFlowVariable.hpp

```diff
@@ -66,6 +66,23 @@ class CFlowVariable : public CVariable {
                 unsigned long nprimvargrad, const CConfig* config);
 
  public:
+  mutable su2vector<int8_t> NonPhysicalEdgeCounter;  /*!< \brief Non-physical reconstruction counter for each edge. */
+  /*!
+   * \brief Updates the non-physical counter of an edge.
+   * \param[in] iEdge - Edge index.
+   * \param[in] isNonPhys - Should be 1 (true) if a non-physical reconstruction was occurred.
+   * \return Whether the reconstruction should be limited to first order, based on the counter.
+   */
+  template <class T>
+  inline T UpdateNonPhysicalEdgeCounter(unsigned long iEdge, const T& isNonPhys) const {
+    if (isNonPhys != 0) {
+      /*--- Force 1st order for this edge for at least 20 iterations. ---*/
+      NonPhysicalEdgeCounter[iEdge] = 21;
```

Comment:
20 iterations ought to be enough for everyone? Or should it be an option that can be set?  In any case, it might be a good idea to move the magic number to a more proper location?

Reply:
🤷 if it is not, it is probably better to just use first 1st order to start, it has been 20 since Tom implemented this. We dont really have a place for this type of constant, at least it is not repeated now.